informational/req-res-reliability.md

```diff
@@ -0,0 +1,89 @@
+
+## Suggestions
+
+### Node health
+
+Node health is a metric meant to determine how reliable a light node is.
+We consider this reliability to be dependant on amount of simultaneous connections to responsive service nodes.
+Unfortunately the more connections light node establishes - the more bandwidth is consumed.
+To address this we suggest following metrics:
+- unhealthy - no connections to service nodes are available regardless of protocol;
+- minimally healthy:
+  - Filter has one service node connection;
+  - LightPush protocol has one service node connection;
+- sufficiently healthy:
+  - Filter has at least 2 connections available to service nodes;
+  - LightPush has at least 2 connections available to service nodes;
+
+### Peers and connection management
+
+#### Pool of reliable service nodes
+Light node should maintain a pool of reliable service nodes for each protocol.
+In case service node fails to serve protocol request from a light node 3 times - light node should drop connection to it and a new service node should be connected and added to the pool instead.
+
+#### Seection of discovered service nodes
+During discovery light node should filter out service nodes based on preferences before establishing connection.
+These preferences might include:
+- [Libp2p multiadresses](https://github.com/libp2p/specs/blob/master/addressing/README.md) of a service node;
+- Waku or libp2p protocols that a service node implements;
+- Wakus shards that a service node is part of;
+
+More details about discovery can be found at [WAKU2 Discovery domain](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/10/waku2.md#discovery-domain) or [RELAY-SHARDING Discovery](https://github.com/waku-org/specs/blob/master/standards/core/relay-sharding.md#discovery).
+
+Examples of filtering:
+- When [Circuit V2](https://github.com/libp2p/specs/blob/master/relay/circuit-v2.md) multiaddresses discovered by a light node - it might avoid connecting such service nodes and wait for service nodes that can be connected directly;
+- When light node discovers service nodes that implement needed Waku protocols - it should prioritize those that implement most recent version of protocol;
+- Light node must connect only to those service nodes that participate in needed shard;
+
+#### Continuous discovery
+Light node must keep information about service nodes up to date.
+For example when a service node is discovered second time,
+we need to be sure to keep connection information up to date in Peer Store.
+
+### Light Push
+
+- While sending message with Light Push - it is advised to use more than 1 peer in order to increase chances of delivering message.
+
+- If sending message is failed to all of the peers - node should try to re-send message after some interval and continue doing so until OK response is received. 
```

Comment:
I think we should take the opportunity to define application concern (user of waku library) vs waku concern (reliablity in waku).

How much should waku res-send? At which point should it be handed over the applicaiton and by extension the end user?

One proposal was a temporal one: Waku owns the message to be sent for ~15s (under the 20s grace period) and attempt during this period to send the message and get peer-to-peer confirmation.

Also, we need to define "sending message failed" similarly to my previous comment: are we saying no response/error response from service node?
Or are we saying we were not able (via store or filter) to confirm the message was handed over to the network?

Reply:
added more definitions, this should be resolved